library/objective-c/EnvoyEngine.m

```diff
@@ -205,16 +205,15 @@ @implementation EnvoyHttpStream {
   EnvoyHttpStream *_strongSelf;
   EnvoyObserver *_platformObserver;
   envoy_observer *_nativeObserver;
-  envoy_stream_t _nativeStream;
-  envoy_engine_t _engineHandle;
+  envoy_stream_t _streamHandle;
 }
 
-- (instancetype)initWithEngine:(envoy_engine_t)engine observer:(EnvoyObserver *)observer {
+- (instancetype)initWithHandle:(envoy_stream_t)streamHandle observer:(EnvoyObserver *)observer {
```

Comment:
I actually like this approach more than the original engine approach

This is a similar concept to what I was thinking about doing for the java side

Definitely can chat more tomorrow in case we think of a better solution